skore/src/skore/item/item_repository.py

```diff
@@ -64,33 +85,48 @@ def get_item(self, key) -> Item:
         Item
             The retrieved item.
         """
-        value = self.storage[key]
-        item_class_name = value["item_class_name"]
-        item_class = ItemRepository.ITEM_CLASS_NAME_TO_ITEM_CLASS[item_class_name]
-        item = value["item"]
+        return ItemRepository.__construct_item(self.storage[key][-1])
 
-        return item_class(**item)
+    def get_items(self, key) -> list[Item]:
```

Comment:
I find it a bit confusing to just add a "s" at the end of `get_item`. What about being really explicit with something like `get_item_history` ?

Reply:
Agreed, it definitely should be more explicit. However, so far `history` has been used to express all past versions, *without* the current one, so it wouldn't work here...

Reply:
I know that @augustebaum don't like it, but we can also use `get_item(key, *, latest=True) -> Union[Item, list[Item]]`.

Reply:
@thomass-dev asks: How about `get_all_items`?

Reply:
Otherwise, let's stick with `get_items`

Reply:
What about `get_item_versions` ?

Reply:
Accepted by all third-parties.